### PR TITLE
Move styling to Input

### LINKED ISSUE #568: Input styling
The input components should have their own styling, not just inherit from `Form` et c.

### PR DESCRIPTION
**Pull request by jensborje**
This PR moves styling relating to input fields from a global mixin called `form-base` to be more component specific. This styling is now in the component `InputBase`. All Input components inherits this.

We are however still to use the global mixin `input-base`, since this is shared with components outside of this scope.

Also removed unnecessary use of `input-base` in some of the input components.

Fix #568

### CODE REVIEW

File: src/components/forms/inputs/SelectInput.scss [hunk 1]
```diff
@@ -1,10 +1,9 @@
-select{
-    @include input-base;
+select {
```
Comment (richardolsson) [inline src/components/forms/inputs/SelectInput.scss]: Is it really a good idea to globally style `select` (and not `.SelectInput`)?

Comment (jensborje) [reply src/components/forms/inputs/SelectInput.scss]: No it isn't. All I did in this commit was to add a space, but we should make improvements when we notice them. Well spotted!

Comment (olleolleolle) [reply src/components/forms/inputs/SelectInput.scss]: 🦅 👁

File: src/components/forms/inputs/SelectInput.jsx [hunk 2]
```diff
@@ -31,10 +31,12 @@ export default class SelectInput extends InputBase {
         }
 
         return (
-            <select value={ this.props.value || '0' }
-                onChange={ this.onChange.bind(this) }>
-                { optionElements }
-            </select>
+            <div className="SelectInput">
```
Comment (olleolleolle) [inline src/components/forms/inputs/SelectInput.jsx]: React-beginner: Just learned about the  `className` property!

https://reactjs.org/docs/faq-styling.html

**Comment by jensborje** [comment]
Updated the PR description. Hope it makes more sense now!

**Pull request merged**